### ISSUE TITLE
[Probe] RelabelingConfigs not been override

### ISSUE THREAD

**Issue by danielcrisap** [prometheus-operator/prometheus-operator#4565]
**What happened?**

Scrape config generated by *Probe* definition hasn’t the `relabel_configs` override by the Probe `relabelingConfigs`, only append to default `relabel_configs` 

**Did you expect to see something different?**

I expect to `relabel_configs` to be overridden by `relabelingConfigs`

**How to reproduce it (as minimally and precisely as possible)**:

Apply the manifest bellow

**Environment**

* Prometheus Operator version: Prometheus-operator:0.54.0

* Kubernetes version information: 1.19

* Kubernetes cluster kind:  EKS

* Manifests:

```yaml
apiVersion: monitoring.coreos.com/v1
kind: Probe
metadata:
  name: blackbox-kubernetes-ingresses
  namespace: monitoring
spec:
  jobName: blackbox-kubernetes-ingresses
  interval: 30s
  module: http_2xx
  prober:
    url: blackbox-prometheus-blackbox-exporter:9115
    scheme: http
    path: /probe
  targets:
    ingress:
      relabelingConfigs:
        - sourceLabels: [__meta_kubernetes_ingress_scheme, __address__, __meta_kubernetes_ingress_path]
          regex: (.+);(.+);(.+)
          replacement: https://${2}/health
          targetLabel: __param_target
```

* Prometheus Operator Logs:

```
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:01:51.219125889Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:01:51.577420224Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:01:53.806054672Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:01:54.857042687Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:01:55.533324079Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:01:56.875676714Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:01:57.523938512Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:04:31.48292563Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:05:49.154525772Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:05:49.527347743Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:06:45.432374855Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:06:45.905671441Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:11:01.079658534Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
prometheus-operator-6fcff478df-7bgfs kube-prometheus-stack level=info ts=2022-02-13T18:11:01.385646873Z caller=operator.go:1218 component=prometheusoperator key=monitoring/prometheus-prometheus msg="sync prometheus"
```

**Anything else we need to know?**:

Result of Probe config

```YAML
- job_name: probe/monitoring/blackbox-kubernetes-ingresses
  honor_timestamps: true
  params:
    module:
    - http_2xx
  scrape_interval: 10s
  scrape_timeout: 10s
  metrics_path: /probe
  scheme: http
  follow_redirects: true
  relabel_configs:
  - source_labels: [job]
    separator: ;
    regex: (.*)
    target_label: __tmp_prometheus_job_name
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: job
    replacement: blackbox-kubernetes-ingresses
    action: replace
  - source_labels: [__meta_kubernetes_ingress_scheme, __address__, __meta_kubernetes_ingress_path]
    separator: ;
    regex: (.+);(.+);(.+)
    target_label: __param_target
    replacement: ${1}://${2}${3}
    action: replace
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_ingress_name]
    separator: ;
    regex: (.*)
    target_label: ingress
    replacement: $1
    action: replace
  - source_labels: [__param_target]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: __address__
    replacement: blackbox-prometheus-blackbox-exporter:9115
    action: replace
  - source_labels: [__meta_kubernetes_ingress_scheme, __address__, __meta_kubernetes_ingress_path]
    separator: ;
    regex: (.+);(.+);(.+)
    target_label: __param_target
    replacement: https://${2}/health
    action: replace
```

**Comment by slashpai** [comment]
Could you please provide prometheus-operator logs as well?

**Comment by simonpasquier** [comment]
Can you provide more details about the ingress objects and your particular use case? IIUC you want to build your own `__param_target` value but given that `__address__` is overridden just before, you can't do that.

In general, I would agree that we lack good documentation for the Probe resource (for instance what is getting scraped if you use ingress targets as you did).

**Comment by danielcrisap** [comment]
> Could you please provide prometheus-operator logs as well?

I update the report with the logs from prometheus-operator

**Comment by danielcrisap** [comment]
> Can you provide more details about the ingress objects and your particular use case? IIUC you want to build your own `__param_target` value but given that `__address__` is overridden just before, you can't do that.
> 
> In general, I would agree that we lack good documentation for the Probe resource (for instance what is getting scraped if you use ingress targets as you did).

I'm trying to use blackbox to validate all my ingress from an external point of view, and all have the `/health` path to be scraped.

And my problem is exactly as you said, the `__address__` is overridden before and I can't use that value.

<img width="2041" alt="Screen Shot 2022-02-13 at 13 18 35" src="https://user-images.githubusercontent.com/32133534/153768890-a6d48c0a-19a0-463b-82dd-d6372a92d319.png">
 
I'd like to have a way to override these `defaults` `relabel_configs`, so I could define my owns.

**Comment by simonpasquier** [comment]
One solution might be to move the relabeling of the `__address__` label at the very end.

```yaml
  relabel_configs:
  - source_labels: [job]
    separator: ;
    regex: (.*)
    target_label: __tmp_prometheus_job_name
    replacement: $1
    action: replace
  - separator: ;
    regex: (.*)
    target_label: job
    replacement: blackbox-kubernetes-ingresses
    action: replace
  - source_labels: [__meta_kubernetes_ingress_scheme, __address__, __meta_kubernetes_ingress_path]
    separator: ;
    regex: (.+);(.+);(.+)
    target_label: __param_target
    replacement: ${1}://${2}${3}
    action: replace
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_ingress_name]
    separator: ;
    regex: (.*)
    target_label: ingress
    replacement: $1
    action: replace
  # user-defined relabeling configs start here.
  - source_labels: [__meta_kubernetes_ingress_scheme, __address__, __meta_kubernetes_ingress_path]
    separator: ;
    regex: (.+);(.+);(.+)
    target_label: __param_target
    replacement: https://${2}/health
    action: replace
  # user-defined relabeling configs end here.
  # then the operator sets the instance label to the final probed target (unless the user already defined a value).
  - source_labels: [__param_target, instance]
    separator: ;
    regex: (.*);
    target_label: instance
    replacement: $1
    action: replace
  # then the operator overrides __address__ with the address of the prober.
  - separator: ;
    regex: (.*)
    target_label: __address__
    replacement: blackbox-prometheus-blackbox-exporter:9115
    action: replace
```

Another option would be to inject a temporary label to remember the original address value (just like we do with the job label).

```yaml
  relabel_configs:
  - source_labels: [__address__]
    separator: ;
    regex: (.*)
    target_label: __tmp_prometheus_address
    replacement: $1
    action: replace
```

**Comment by simonpasquier** [comment]
closed by #4598

**Comment by danielcrisap** [comment]
Issue closed due to the solution already applied in the version [v0.55.0](https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.55.0)

Thanks, @simonpasquier, and @mikechengwei  for the implementation !!
